internal/clients/index.go

```diff
@@ -78,6 +78,69 @@ func (a *ApiClient) DeleteElasticsearchIlm(policyName string) diag.Diagnostics {
 	return diags
 }
 
+func (a *ApiClient) PutElasticsearchComponentTemplate(template *models.ComponentTemplate) diag.Diagnostics {
+	var diags diag.Diagnostics
+	templateBytes, err := json.Marshal(template)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	log.Printf("[TRACE] sending request to ES: %s to create component template '%s' ", templateBytes, template.Name)
+
+	res, err := a.es.Cluster.PutComponentTemplate(template.Name, bytes.NewReader(templateBytes))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	defer res.Body.Close()
+	if diags := utils.CheckError(res, "Unable to create component template"); diags.HasError() {
+		return diags
+	}
+
+	return diags
+}
+
+func (a *ApiClient) GetElasticsearchComponentTemplate(templateName string) (*models.ComponentTemplateResponse, diag.Diagnostics) {
+	var diags diag.Diagnostics
+	req := a.es.Cluster.GetComponentTemplate.WithName(templateName)
+	res, err := a.es.Cluster.GetComponentTemplate(req)
+	if err != nil {
+		return nil, diag.FromErr(err)
+	}
+	defer res.Body.Close()
```

Comment:
```suggestion
	defer res.Body.Close()
	if res.StatusCode == http.StatusNotFound {
		return nil, nil
	}
```

We also need to handle situation when the component was deleted manually, but still has record in the TF state and in the TF config files. 

See also template, when the removed template was handled same way: https://github.com/elastic/terraform-provider-elasticstack/blob/68ba59435c4166b6f929345d7f506811e24f8fa8/internal/clients/index.go#L172-L174

Also see my other suggestion, when the check is done and ID of the resource set to empty string (`""`) to force the  re-creation of the resource